### PR TITLE
Rubicon Bid Adapter: add outstream rendering

### PR DESCRIPTION
**Pull request by smozhaiskyi-rubi**
## Type of change
- [x] Feature

## Description of change
Render outstream ads with Rubicon player.

The renderer appearance can be configured with the following parameters, all of them are optional:
```
pbjs.setConfig({
  rubicon: {
    rendererConfig: {
      align: 'center',   // player placement: left|center|right
      position: 'append'   // player position relative to ad unit: append|prepend|before|after
      closeButton: false,   // display 'Close' button
      label: '',   // custom text to display above the player
      collapse: true   // remove the player from the page after ad playback
    }
  }
});
```

### CODE REVIEW

**Comment by Rothalack** [comment]
It seems to be working properly for me. The dev docs update should be the only thing left to do. I think the minimum you'd need is to add those new params. Maybe your Outstream section could use an update, but that's up to you.

https://github.com/prebid/prebid.github.io/blob/master/dev-docs/bidders/rubicon.md

**Review by mmoschovas — APPROVED** [review]
LGTM

**Review by Rothalack — APPROVED** [review]
Just waiting on dev docs update.

**Review by bretg — CHANGES_REQUESTED** [review]
We need to know the open source repo location where https://video-outstream.rubiconproject.com/apex-2.0.0.js lives and can be inspected

Please add a comment?

**Comment by smozhaiskyi-rubi** [comment]
The source code for Apex2 outstream ad unit renderer is available for inspection here: https://github.com/rubicon-project/apex2

**Comment by bretg** [comment]
Thanks @smozhaiskyi-rubi . Now we need someone willing to confirm the external renderer code.

**Comment by patmmccann** [comment]
I think we also need a docs update on how to set these parameters in the bidder parameters doc

File: modules/rubiconBidAdapter.js [hunk 1]
```diff
@@ -783,6 +791,63 @@ function _renderCreative(script, impId) {
...
+  if (el) {
+    el.style.setProperty('display', 'none');
+  }
+}
+
+function hideSmartAdServerIframe(adUnit) {
+  const el = adUnit.querySelector("script[id^='sas_script']");
+  if (el && el.nextSibling && el.nextSibling.localName === 'iframe') {
+    el.nextSibling.style.setProperty('display', 'none');
+  }
```
Comment (karimMourra) [inline modules/rubiconBidAdapter.js:805]: for better minification i recommend using a `nextSibling` var
```
const nextSibling = el && el.nextSibling;
if (nextSibling && nextSibling.localName === 'iframe') {
    nextSibling.style.setProperty('display', 'none');
}
```

this should result in fewer chars and reduce file size

Comment (smozhaiskyi-rubi) [reply modules/rubiconBidAdapter.js]: Done.

**Review by karimMourra — APPROVED** [review]

**Review by bretg — APPROVED** [review]

**Comment by bretg** [comment]
Fixed conflict

**Comment by bretg** [comment]
Docs done. Already merged.

File: test/spec/modules/rubiconBidAdapter_spec.js [hunk 2]
```diff
@@ -3050,6 +3050,153 @@ describe('the rubicon adapter', function () {
...
+            placement: {
+              align: 'left',
+              attachTo: '#outstream_video1_placement',
+              position: 'append',
+            },
+            vastUrl: 'https://test.com/vast.xml',
+            width: 640
+          });
+          // cleanup
+          adUnit.remove();
```
Comment (patmmccann) [inline test/spec/modules/rubiconBidAdapter_spec.js:3196]: this is not ie 11 compatible

Comment (robertrmartinez) [reply test/spec/modules/rubiconBidAdapter_spec.js:3196]: 🤦

Comment (patmmccann) [reply test/spec/modules/rubiconBidAdapter_spec.js:3196]: https://github.com/prebid/Prebid.js/blob/804295aa2dae0484d67891b73fcfc401ef8244f1/test/spec/modules/videoNowBidAdapter_spec.js#L6

Comment (patmmccann) [reply test/spec/modules/rubiconBidAdapter_spec.js:3196]: that polyfill from the other adapter is why your tests passed before. removing it causes this to fail.

Comment (robertrmartinez) [reply test/spec/modules/rubiconBidAdapter_spec.js:3196]: ah ok, thought this was merged with failing unit tests.

I prefer to fix this by not using a polyfill.

I will put a PR together

Comment (robertrmartinez) [reply test/spec/modules/rubiconBidAdapter_spec.js:3196]: @patmmccann Should this PR be made in master or 5.0 branch?

Guess it does not matter?

Comment (harpere) [reply test/spec/modules/rubiconBidAdapter_spec.js:3196]: let's do it in *master*, then *master* should be merged back into *5.0*

Comment (patmmccann) [reply test/spec/modules/rubiconBidAdapter_spec.js:3196]: works for me

**Pull request merged**